src/foam/dao/EasyDAO.js

```diff
@@ -566,35 +566,41 @@ return delegate;
       name: 'testData',
       generateJava: false
     },
+    {
+      documentation: 'Enables automated adding of aware doas to qualifying decorator chains',
+      name: 'enableAware',
+      class: 'Boolean',
+      value: true
+    },
     {
       name: 'serviceProviderAware',
       class: 'Boolean',
-      javaFactory: 'return foam.nanos.auth.ServiceProviderAware.class.isAssignableFrom(getOf().getObjClass());'
+      javaFactory: 'return !getEnableAware() ? false : foam.nanos.auth.ServiceProviderAware.class.isAssignableFrom(getOf().getObjClass());'
```

Comment:
```suggestion
      javaFactory: 'return getEnableAware() && foam.nanos.auth.ServiceProviderAware.class.isAssignableFrom(getOf().getObjClass());'
```

is simpler. Please change the others as well.